ServerCore/Areas/Identity/UserAuthorizationPolicy/AuthorizationHelper.cs

```diff
@@ -60,13 +61,26 @@ public static async Task<Team> GetTeamFromContext(AuthorizationHandlerContext co
             return null;
         }
 
+        public static EventRole GetEventRoleFromContext(AuthorizationHandlerContext context)
+        {
+            if (context.Resource is AuthorizationFilterContext filterContext)
+            {
+                string eventRole = filterContext.RouteData.Values["eventRole"] as string;
+
+                Enum.TryParse<EventRole>(eventRole, out EventRole role);
```

Comment:
If it's not in the enum, I think you'll get admin (because it's 0). You should probably check the result of this.